File: 프로그래머스/1/155652. 둘만의 암호/둘만의 암호.py
def solution(s, skip, index):
    answer = ''
    alphabet = [chr(i) for i in range(97, 123)]
    for r in skip:
        alphabet.remove(r)
        
    for i in s:
        locate = alphabet.index(i)
        idx = locate + int(index)
        while idx >= len(alphabet): 
            idx -= len(alphabet)
        answer += alphabet[idx]
        
    return answer

File: 프로그래머스/1/155652. 둘만의 암호/둘만의 암호.java
import java.util.*;

class Solution {
    public String solution(String s, String skip, int index) {
        String answer = "";
        Character[] alpha = {'a', 'b', 'c', 'd', 'e', 'f', 'g', 'h', 'i', 'j', 'k', 'l', 'm', 'n', 'o', 'p', 'q', 'r', 's', 't', 'u', 'v', 'w', 'x', 'y', 'z'};
        List<Character> list = new ArrayList<>();
        
        for(int i = 0; i < alpha.length; i++) 
            list.add(Character.valueOf(alpha[i])); 
        
        for(int i = 0; i < skip.length(); i++) 
            list.remove(Character.valueOf(skip.charAt(i)));
        
        int size = list.size();
        for(int i = 0 ; i < s.length(); i++){
            int idx = list.indexOf(s.charAt(i)) + index;
            idx %= size;
            answer += Character.valueOf(list.get(idx));
        }
        return answer;
    }
}